### PR TITLE
Aa + Initializers + cast + string switch + utf decoding

### PR DESCRIPTION
**Pull request by MrcSnm**
Associative array implementation is working

Array initializer fixed:
float[] f = new float[2]; assert(float[0] is float.init) Struct Array initializer fixed.
Interface cast
assert message now showing the message

Added missing properties for delegate and const.
Added version for CarelessAllocation which will enable every allocation that should require a GC. inline_concat version.

Added utf foreach(dchar ch) decoding

### CODE REVIEW

File: arsd-webassembly/std/stdio.d [hunk 1]
```diff
@@ -4,14 +4,6 @@ import arsd.webassembly;
...
-		var str = "";
-		for(var i = 0; i < arguments.length; i++)
-			str += arguments[i];
-
-		str += "\n";
-
-		var txt = document.createTextNode(str);
-		var fd = document.getElementById("stdout");
-		fd.appendChild(txt);
+		console.log.apply(null, arguments);
```
Comment (adamdruppe) [inline arsd-webassembly/std/stdio.d]: omg you broke my writeln lolololol

File: servewasm.py [hunk 2]
```diff
@@ -0,0 +1,17 @@
+#!/usr/bin/env python3
+
+import http.server
```
Comment (adamdruppe) [inline servewasm.py]: WTF why is there python trash in the D repo?!?!?!!?!?!?!?!!?!?!!?!?!!?!!?

**Review by adamdruppe — CHANGES_REQUESTED** [review]
death to python

File: arsd-webassembly/std/stdio.d (hunk 1 above)
Comment (MrcSnm) [reply arsd-webassembly/std/stdio.d]: Oh srry gonna revert it

**Pull request merged**